docs/usage/enabling_immutable_snapshots.md

```diff
@@ -0,0 +1,368 @@
+# Enabling Immutable Snapshots in `etcd-backup-restore`
+
+This guide walks you through the process of enabling immutable snapshots in `etcd-backup-restore` by leveraging bucket-level immutability features provided by cloud storage providers like Google Cloud Storage (GCS) and Azure Blob Storage (ABS). Enabling immutability ensures that your backups are tamper-proof and comply with regulatory requirements.
+
+---
+
+## Terminology
+
+- **Bucket / Container**: A storage resource in cloud storage services where objects (such as snapshots) are stored. GCS uses the term **bucket**, while ABS uses **container**.
+
+- **Retention Policy**: A configuration that specifies a minimum period during which objects in a bucket/container are protected from deletion or modification.
+
+- **Retention Period**: The duration defined by the retention policy during which objects remain immutable.
+
+- **Immutability**: A state where objects cannot be modified or deleted until the retention period expires.
+
+- **Locking**: The action of making a retention policy permanent, preventing any reduction or removal of the retention period.
+
+- **ETag**: An identifier representing a specific version of a policy or object, used for concurrency control.
+
+---
+
+## Overview
+
+Currently, `etcd-backup-restore` supports bucket-level immutability for GCS and ABS. Support for AWS S3 is planned for a future release.
+
+- **Retention Policy**: You can add a retention policy to a bucket/container to specify a retention period.
+  - When a retention policy is set, objects in the bucket/container can only be deleted or replaced once their age exceeds the retention period.
+  - The policy retroactively applies to existing objects as well as new objects added to the bucket/container.
+
+- **Locking a Retention Policy**: You can lock a bucket's/container's retention policy to permanently enforce it.
+  - Once locked, you cannot remove the retention policy or reduce its retention period.
+  - You cannot delete a bucket/container with a locked retention policy unless every object has met the retention period.
+  - You can increase the retention period of a locked policy if needed.
+
+> **Caution**: Locking a retention policy is an irreversible action.
+
+---
+
+## Configure Bucket-Level Immutability
+
+By configuring a retention policy on your storage bucket/container, you ensure that all snapshots are stored in an immutable (Write Once, Read Many) state for a specified duration. This prevents snapshots from being modified or deleted until they reach the end of the retention period.
+
+### Configure a Retention Policy
+
+You can set a time-based retention policy on your bucket/container. The retention policy specifies the minimum duration for which the objects must remain immutable.
```

Comment:
Can a line suggesting configuring the policy is also achievable in the cloud provider's respective console/portal, the way this doc currently suggests for adding tags to objects?